api/handlers/service_instance_test.go

```diff
@@ -73,6 +73,294 @@ var _ = Describe("ServiceInstance", func() {
+						}, nil)
+					})
+
+					It("includes broker fields in the response", func() {
+						Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+						Expect(rr).To(HaveHTTPBody(SatisfyAll(
+							MatchJSONPath("$.included.service_brokers[0].guid", "service-broker-guid"),
+							MatchJSONPath("$.included.service_brokers[0].name", "service-broker-name"),
+						)))
+					})
+				})
+			})
+		})
+	})
+
+	Describe("GET /v3/service_instances/:guid/credentials", func() {
+		BeforeEach(func() {
+			serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+				GUID: "service-instance-guid",
+				Type: korifiv1alpha1.UserProvidedType,
+			}, nil)
+
+			serviceInstanceRepo.GetServiceInstanceCredentialsReturns(map[string]any{
+				"foo": "bar",
+			}, nil)
+
+			reqPath += "/service-instance-guid/credentials"
+		})
+
+		It("gets the service instance credentials", func() {
+			Expect(serviceInstanceRepo.GetServiceInstanceCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualServiceInstanceGUID := serviceInstanceRepo.GetServiceInstanceArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualServiceInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(serviceInstanceRepo.GetServiceInstanceCredentialsCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+			Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
+			Expect(rr).To(HaveHTTPBody(SatisfyAll(
+				MatchJSONPath("$.foo", "bar"),
+			)))
+		})
+
+		When("the service instance is not user-provided", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.ManagedType,
+				}, nil)
+			})
+
+			It("returns a 404 Not Found error", func() {
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("the service instance does not have credentials", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.UserProvidedType,
+				}, nil)
+
+				serviceInstanceRepo.GetServiceInstanceCredentialsReturns(map[string]any{}, apierrors.NewNotFoundError(nil, repositories.ServiceInstanceResourceType))
+			})
+
+			It("returns 404 Not Found", func() {
+				_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
+				Expect(actualAuthInfo).To(Equal(authInfo))
+				Expect(actualInstanceGUID).To(Equal("service-instance-guid"))
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("getting the service instance fails with not found", func() {
```

Comment:
the handler should only interpret the "forbidden" error and present it as not found. Any other sort of errors (regiardless not found, general error, whatever) should be just returned. 

Having said that, I would alter this `When` to verify that when the repository returns a generic error, it results into an unexpected error (HTTP 500). In other words, as the handler does not handle the not found error in any particular way, I would go for the least specific test case.